job_executor/worker/build_dataset_worker.py

```diff
@@ -71,12 +84,14 @@ def run_worker(job_id: str, dataset_name: str, logging_queue: Queue):
         transformed_metadata = Metadata(**transformed_metadata_json)
 
         temporality_type = transformed_metadata.temporality
-
-        job_service.update_job_status(job_id, "pseudonymizing")
-        pseudonymized_data_path = dataset_pseudonymizer.run(
-            validated_data_file_path, transformed_metadata, job_id
-        )
-        local_storage.delete_working_dir_file(validated_data_file_path)
+        if _dataset_requires_pseudonymization(input_metadata):
+            job_service.update_job_status(job_id, "pseudonymizing")
+            pseudonymized_data_path = dataset_pseudonymizer.run(
+                validated_data_file_path, transformed_metadata, job_id
+            )
+            local_storage.delete_working_dir_file(validated_data_file_path)
+        else:
+            pseudonymized_data_path = validated_data_file_path
```

Comment:
I agree that it is somewhat confusing. The issue is that when we actually pseudonymize we set the path to the new pseudonymized file:
```py
pseudonymized_data_path = dataset_pseudonymizer.run(
    validated_data_file_path, transformed_metadata, job_id
)
```

and then in the next step we use that path:
```py
dataset_partitioner.run(pseudonymized_data_path, dataset_name)
```

so now if we use another name for the path that has passed the pseudonymizing step without needing pseudonymizing, we end up having to do if statements to check if it got pseudonymized or not throughout the rest of the code.

Let me have a look and think about it over lunch 👍🏻 (and post ideas if you are thinking of something 🙇🏻 )

Reply:
Could an explanatory comment work here maybe ?
```py
# No columns required pseudonymization so using validated data path 
```

Reply:
I think this could written be like that:
```
if _dataset_requires_pseudonymization(input_metadata):
    ...
    data_path = dataset_pseudonymizer.run(
        validated_data_file_path, transformed_metadata, job_id
     )
     ...
else:
    data_path = validated_data_file_path
# the rest of the code can now use data_path variable
```

Reply:
doesn't that feel slightly generic and confusing too? the file produced after the validator step is called "validated_data_path" and the one after pseudonymizer step is called "data_path" 🤔 
I guess I can name both data_path and just reassign after each step? that might be the most intuitive ill try 👍🏻